microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -49,14 +49,18 @@ type GMConnectorReconciler struct {
 func getKubeConfig() (*rest.Config, error) {
 	var config *rest.Config
 	var err error
-
 	if _, err = os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/token"); err == nil {
 		config, err = rest.InClusterConfig()
 		if err != nil {
 			return nil, fmt.Errorf("failed to create in-cluster config: %w", err)
 		}
 	} else {
 		kubeconfigPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+		if _, err := os.Stat(kubeconfigPath); os.IsNotExist(err) {
```

Comment:
Unit tests need to be reproducible, running on our laptop or in a real cluster, so what is in that KubeConfig matters. Would it make sense to have a canned/static KubeConfig for unit test purposes

Reply:
Let's use fakeclient from k8s for GMC UT.    You can reference:      k8s.io/client-go/kubernetes/fake

Reply:
@mkbhanda that's what I do in this PR, I use a static but fake KubeConfig for the non-k8s env to do unit tests.
@irisdingbj we cannot use fakeclient here in the real code.